### ISSUE TITLE
Problem with formatting union expressions when using `value_counts`

### ISSUE THREAD

**Issue by kesmit13** [ibis-project/ibis#4636]
I'm working on a subclass of the MySQL backend and using unions. When attempting to do a `value_counts` on a union, I get an attribute error. Here is a simple test using our backend (this DataFrame upload might not work in the actual MySQL, but should be fairly simple to replicate).
```
df = pd.DataFrame([[0, 1], [2, 3]], columns=['a', 'b'])
tbl = conn.create_table('test_union', df, force=True)
u = ibis.union(tbl, tbl)
u.a.value_counts()
```

Here is the tail end of the exception.
```
...
~/.pyenv/versions/3.9.4/lib/python3.9/site-packages/ibis/expr/format.py in _fmt_value_expr(expr, aliases)
    555     Forwards the call on to the specific operation dispatch rule.
    556     """
--> 557     return fmt_value(expr.op(), aliases=aliases)
    558 
    559 

~/.pyenv/versions/3.9.4/lib/python3.9/functools.py in wrapper(*args, **kw)
    875                             '1 positional argument')
    876 
--> 877         return dispatch(args[0].__class__)(*args, **kw)
    878 
    879     funcname = getattr(func, '__name__', 'singledispatch function')

~/.pyenv/versions/3.9.4/lib/python3.9/site-packages/ibis/expr/format.py in _fmt_value_table_node(op, aliases, **_)
    669     if not hasattr(op, 'table'):
    670         import pdb; pdb.set_trace()
--> 671     return f"{aliases[op.table.op()]}"
    672 
    673 

AttributeError: 'Union' object has no attribute 'table'
```

**Comment by jcrist** [comment]
Thanks for opening this, this is a bug in the expr repr (the error comes from repring `u.a.value_counts()`, not the generation of the expression). If you avoid ever repr'ing the expression, or enable interactive mode

```
ibis.options.interactive = True
```

you should be able to work around this for now.

Should be an easy fix though, I can look into this later.

**Comment by jcrist** [comment]
Should be fixed by #4637.
